### PR TITLE
k8sbundle notifier: Docs cleanup and bug fix

### PR DESCRIPTION
**Pull request by faisal-memon**
Signed-off-by: Faisal Memon <f.memon@f5.com>

**Pull Request check list**

- [x] Commit conforms to CONTRIBUTING.md?
- [x] Proper tests/regressions included?
- [x] Documentation updated?

**Affected functionality**
k8sbundle notifier

**Description of change**
~~With previous PRs to update k8sbundle notifier to rotate webhooks and apiservices, we forgot to add an option to disable ConfigMap rotation. So that, for example, you could rotate webhook caBundles without setting up the ConfigMap.~~

~~This PR adds a `config_map_enable` flag to control whether or not to rotate the bundle in the ConfigMap. The default value is true to preserve backwards compatibility.~~

Contains:
- Some docs cleanup
- Bug fix where cancelWatcher is not being setup for APIService

### CODE REVIEW

**Comment by faisal-memon** [comment]
> Hey @faisal-memon. Thanks for opening this. I think if we go down this route, I'd rather the `config_map` configurable become optional instead of introducing a new `config_map_enable` configurable.
> 
> We should then fail to Configure if neither config_map, webhook, or service is configured for updates.
> 
> Just so I understand the use case, do you have something you are using instead to provide the bootstrap bundle to agents other than the ConfigMap, but still using this to update the webhook and service configs?

Hi @azdagron. Thanks for looking at this. I was working on a writeup for how to use SPIRE to manage webhook certificates and was using the [simple_sat](https://github.com/spiffe/spire-examples/tree/master/examples/k8s/simple_sat) example as a base. That example doesn't use the k8sbundle notifier so as i was modifying the example I noticed that I couldn't just have it rotate the webhook certs, I had to also set up the configmap rotation. Logically it seemed like i should have a path to just rotate webhook certs without the cabundle. So that's how i came to this PR, not so much a use case but just that it seemed weird that the configmap portion can't be disabled but the webhook and apiservice can be. 

Open to structuring it differently, or closing it off if you don't think its an issue. I just picked this approach for backwards compatibility.

**Comment by azdagron** [comment]
Oh right. `config_map` IS optional today and gets a default value if unset. I don't know if other maintainers have opinions here, but I'd probably lean towards not changing it unless there is a clear use case. @evan2645, any thoughts here?

**Comment by faisal-memon** [comment]
> Oh right. `config_map` IS optional today and gets a default value if unset. I don't know if other maintainers have opinions here, but I'd probably lean towards not changing it unless there is a clear use case. @evan2645, any thoughts here?

Removed the `config_map_enable` flag. Bug fix and docs cleanup still remain. What do you think of updating the simple_sat/simple_psat examples to use the k8sbundle notifier?

File: pkg/server/plugin/notifier/k8sbundle/k8sbundle.go [hunk 1]
```diff
@@ -192,7 +192,7 @@ func (p *Plugin) setConfig(config *pluginConfig) error {
 		p.cancelWatcher()
 		p.cancelWatcher = nil
 	}
-	if config.WebhookLabel != "" {
+	if config.WebhookLabel != "" || config.APIServiceLabel != "" {
```
Comment (azdagron) [inline pkg/server/plugin/notifier/k8sbundle/k8sbundle.go:195]: Super curious how you noticed this, as SPIRE itself currently only configures once...?

Comment (faisal-memon) [reply pkg/server/plugin/notifier/k8sbundle/k8sbundle.go:195]: I was making the change to add that new flag and noticed this bit was missing.

**Review by azdagron — APPROVED** [review]
Thanks,  @faisal-memon !

**Pull request merged**